qiskit_experiments/library/characterization/multi_state_discrimination.py

```diff
@@ -0,0 +1,121 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Multi state discrimination experiment."""
+
+from typing import Dict, List, Optional
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Gate
+from qiskit.providers import Backend
+from qiskit.providers.options import Options
+from qiskit.pulse import ScheduleBlock
+from qiskit.qobj.utils import MeasLevel
+from qiskit_experiments.framework import BaseExperiment
+from qiskit_experiments.library.characterization import MultiStateDiscriminationAnalysis
+
+
+class MultiStateDiscrimination(BaseExperiment):
+    r"""An experiment that prepares the first :math:`n+1` energy states,
+    i.e. :math:`|0\rangle,\cdots,|n\rangle`. TODO:is the name ok? not discriminating yet.
+
+    # section: example
+
+        The steps to run a multi state discrimination experiment are TODO
+
+    # section: reference
+        `Qiskit Textbook <https://qiskit.org/textbook/ch-quantum-hardware/accessing\
+        _higher_energy_states.html>`_.
+
+    # section: tutorial
+        :doc:`TODO?`
```

Comment:
That would be nice. Does not need to be too long.

Reply:
moved tutorial to #936